### ISSUE TITLE
[EUI] Remove `@osd/ui-framework` package and replace KUI with EUI

### ISSUE THREAD

**Issue by tmarkley** [opensearch-project/OpenSearch-Dashboards#1060]
Visualizations, Timeline, Discover are just some of the areas of the code base that reference KUI and the other UI Framework components, but this package is related to multiple security vulnerabilities. We need to get rid of it. I will add more details as we dive deeper.

Related to #695

**Comment by joshuarrrr** [comment]
@tmarkley Can this be closed thanks to #1105?

**Comment by tmarkley** [comment]
> @tmarkley Can this be closed thanks to #1105?

No, we still have to replace all of the usages of the KUI components with EUI (or the new design system when we fork) components. See https://github.com/opensearch-project/OpenSearch-Dashboards/blob/89d3872f314aedf9c4f281ea73f784a5802e0175/packages/osd-ui-framework/src/components/index.js#L31-L92

**Comment by tmarkley** [comment]
Related to https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1560. @ananzh are you working on replacing usages of KUI with OUI as you're removing Angular?

**Comment by joshuarrrr** [comment]
@tmarkley I made some related subtask issues to better track. And yes, we'll try to opportunistically clean these up when we migrate, deprecate, or rewrite the affected plugins.
